src/Spice86/Views/DisassemblyView.axaml.cs

```diff
@@ -1,57 +1,85 @@
+    
+    private void DisassemblyView_AttachedToVisualTree(object? sender, Avalonia.VisualTreeAttachmentEventArgs e) {
+        // Activate the view model when the view is attached to the visual tree
+        _viewModel?.Activate();
+    }
+
+    private void DisassemblyView_DetachedFromVisualTree(object? sender, Avalonia.VisualTreeAttachmentEventArgs e) {
+        // Deactivate the view model when the view is detached from the visual tree
+        _viewModel?.Deactivate();
+    }
 
-    private void StartAddressTextBox_KeyUp(object? sender, KeyEventArgs e) {
-        ExecUpdateDisassemblyCommand(e, true);
+    private static void ViewModel_PropertyChanged(object? sender, PropertyChangedEventArgs e) {
     }
 
-    private void ExecUpdateDisassemblyCommand(KeyEventArgs e, bool isUsingLinearAddressing) {
-        if (DataContext is DisassemblyViewModel viewModel && e.Key == Key.Enter &&
-            viewModel.UpdateDisassemblyCommand.CanExecute(null)) {
-            viewModel.UpdateDisassemblyCommand.Execute(null);
+    private void OnBreakpointClicked(object? sender, TappedEventArgs e) {
+        if (sender is Control {DataContext: DebuggerLineViewModel debuggerLine}) {
+            _viewModel?.ToggleBreakpointCommand.Execute(debuggerLine);
+            e.Handled = true;
         }
     }
 
-    private void DisassemblyDataGrid_KeyUp(object? sender, KeyEventArgs e) {
-        if (DataContext is DisassemblyViewModel viewModel) {
-            if(e.Key == Key.C && e.KeyModifiers == KeyModifiers.Control &&
-                viewModel.CopyLineCommand.CanExecute(null)) {
-                viewModel.CopyLineCommand.Execute(null);
-            }
-            if (e.Key == Key.F2 &&
-                viewModel.CreateExecutionBreakpointHereCommand.CanExecute(null)) {
-                viewModel.CreateExecutionBreakpointHereCommand.Execute(null);
-            }
-            if (e.Key == Key.Delete &&
-                viewModel.RemoveExecutionBreakpointHereCommand.CanExecute(null)) {
-                viewModel.RemoveExecutionBreakpointHereCommand.Execute(null);
-            }
+    /// <summary>
+    /// Event handler for when the function selection AutoCompleteBox gets focus.
+    /// </summary>
+    /// <param name="sender">The sender object.</param>
+    /// <param name="e">The focus event arguments.</param>
+    private void OnFunctionSelectionFocus(object? sender, GotFocusEventArgs e) {
+        if (sender is AutoCompleteBox autoCompleteBox) {
+            // Clear the text when the AutoCompleteBox gets focus
+            autoCompleteBox.Text = string.Empty;
         }
     }
 
-    private void FunctionComboBox_SelectionChanged(object? sender, SelectionChangedEventArgs e) {
-        if(this.DataContext is DisassemblyViewModel viewModel) {
-            viewModel.GoToFunctionCommand.Execute(FunctionComboBox.SelectedItem);
+    private void OnFunctionSelectionKeyDown(object? sender, KeyEventArgs e) {
+        if (e.Key == Key.Enter && _viewModel != null) {
```

Comment:
Check for CanExecute first on the Command API.

Reply:
Done.